packages/react-scripts/template/src/App.css

```diff
@@ -1,3 +1,13 @@
+/**
+ * This is a good place to put styles related to the <App /> component.
+ * We recommend that, when possible, a component CSS file only includes styles
+ * that are used by that specific component. This is why all class names start
+ * with "App" in this example. Instead of defining nested selectors for child
+ * components, we recommend passing all relevant information as props (such as
+ * <Button color="red" size="large" />) in React. Then you can have <Button />
```

Comment:
In this case it seems like you are going to directly inline all those styles. One common pattern is to pass down high level props like size={"large" | "medium" | "small"} and then use those to add the class App-size-large, App-size-medium...

I would expand a tiny bit your "nested selector" explanation to explain what I mentioned above. I think this should cover a lot of use cases.